generator.go

```diff
@@ -575,6 +580,38 @@ func tsPrintDefault(v cue.Value) (bool, string, error) {
 	return false, result, nil
 }
 
+func isCueObjConcrete(obj cue.Value) (bool, error) {
+	// when object is concrete, we still need to check the subelement of struct or list is concrete,
+	// otherwise we correct the responds on function IsConcrete for the false positive
+	ok := obj.IsConcrete()
+	if ok {
+		var iter cue.Iterator
+		var err error
+		if obj.Kind() == cue.ListKind {
+			iter, err = obj.List()
+		} else if obj.Kind() == cue.StructKind {
+			var eleiter *cue.Iterator
+			eleiter, err = obj.Fields()
+			iter = *eleiter
+		}
+
+		if err != nil {
+			return ok, err
+		}
+		for iter.Next() {
+			isSubEleConcret, err := isCueObjConcrete(iter.Value())
```

Comment:
nit: variable naming.

This is more art than science, but a good way i've heard it described is, variable names should get more detailed as the references to them get further away from their declarations. The only references to this are within the same control block, within a few lines of code; it's preferable to use a shorter, even not-particularly-meaningful name.